### PR TITLE
Jingyi edit header message permission

### PR DESCRIPTION
**Pull request by Jingyii800**
# Description
Gives the user permission to edit the message displayed in the header.

## Related PRS (if any):
To test this backend PR you need to checkout the [#2345](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2345) frontend PR.
…

## Main changes explained:
- create a new initial permission for editing header message
- Add permission check before running backend logic
- 
## How to test:
1.check into current branch
2.do npm install npm run build and npm start to run this PR locally
3.Clear site data/cache
4. then Please refer to the frontend pull request

### CODE REVIEW

**Review by Kavil-Jain-514 — APPROVED** [review]
The functionality works as expected.
A volunteer user with edit header message permission can edit the error message without any errors or warnings.
image
image

**Review by Logeshwari-Renu — APPROVED** [review]
I have tested the PR and it works as expected. I have left a comment on the front-end [PR](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2345#pullrequestreview-2113583902)

**Comment by akshit0211** [comment]
I left a comment on the front-end PR. It is working as expected. [Review](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2345#pullrequestreview-2117296163)

**Review by akshit0211 — APPROVED** [review]

**Review by nathcarnevalli — APPROVED** [review]
Left my review on [frontend](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2345#pullrequestreview-2119429547).

**Review by tianyangL11 — APPROVED** [review]
left a comment on the front-end PR. It is working as expected.

**Review by jinxyou — APPROVED** [review]
I left a review at https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2345#pullrequestreview-2120835394

File: src/controllers/ownerMessageController.js [hunk 1]
```diff
@@ -15,7 +18,10 @@ const ownerMessageController = function (OwnerMessage) {
   };
 
   const updateOwnerMessage = async function (req, res) {
-    if (req.body.requestor.role !== 'Owner') {
+    if (
+      req.body.requestor.role !== 'Owner' &&
+      !(await helper.hasPermission(req.body.requestor, 'editHeaderMessage'))
+    ) {
```
Comment (nathanah) [inline src/controllers/ownerMessageController.js:24]:
```suggestion
    if (
      !(await helper.hasPermission(req.body.requestor, 'editHeaderMessage'))
    ) {
```
I think the permissions are supposed to be replacing role logic entirely. You add 'editHeaderMessage' in `createInitialPermissions.js`, so Owners should always have this permission. Not requiring them to be an owner allows for the header message to perhaps be delegated to a non-owner account.

File: src/controllers/ownerMessageController.js [hunk 2]
```diff
@@ -40,7 +46,10 @@ const ownerMessageController = function (OwnerMessage) {
   };
 
   const deleteOwnerMessage = async function (req, res) {
-    if (req.body.requestor.role !== 'Owner') {
+    if (
+      req.body.requestor.role !== 'Owner' &&
+      !(await helper.hasPermission(req.body.requestor, 'editHeaderMessage'))
+    ) {
```
Comment (nathanah) [inline src/controllers/ownerMessageController.js:52]:
```suggestion
    if (
      !(await helper.hasPermission(req.body.requestor, 'editHeaderMessage'))
    ) {
```
Same as other comment

**Review by nathanah — CHANGES_REQUESTED** [review]

File: src/controllers/ownerMessageController.js (hunk 2 above)
Comment (Jingyii800) [reply src/controllers/ownerMessageController.js:51]: Thank you, Nathan! I will fix it

**Comment by sahithidhulipala9** [comment]
Left review at https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2345#pullrequestreview-2120835394

**Review by sahithidhulipala9 — APPROVED** [review]

**Review by vishavk1992 — COMMENTED** [review]
it works as expected and i left comment on frontend PR

**Review by nikhilpittala16 — APPROVED** [review]
I have tested the PR and it works as expected. I have left a PR review on the front-end. link can be found here: https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2345#pullrequestreview-2161344009

**Review by Rishitha1309 — APPROVED** [review]
Left my review in the front-end PR. Everything worked as expected.

**Review by Annezhang673 — APPROVED** [review]
I have left a review on the frontend PR#2345. Function works as expected.

**Review by SMitta11 — APPROVED** [review]
I left a comment on the front-end PR. It is working as expected.
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2345#pullrequestreview-2258018110

**Review by AJAYINAVOLU — APPROVED** [review]
Reviewed this PR on the frontend [repo](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2345#pullrequestreview-2278947116).

**Review by nathanah — APPROVED** [review]

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**